### DISCUSSION TITLE
UNETR self-supervise with arbitrary size input

### DISCUSSION THREAD

**Discussion by YingdiZhang** [Project-MONAI/tutorials#583] [Q&A] [answered]
Thanks for the great job.
I am using the tutorial [https://github.com/Project-MONAI/tutorials/blob/master/self_supervised_pretraining/ssl_script_train.py](url) for self-supervised 3D segmentation training with UNETR.  While it seems that the VitAutoEnc requires only cubic input. Can I adjust the input with different （x，y，z）size？

**Comment by Nic-Ma** [comment]
Hi @finalelement ,

Could you please help share some comments here?

Thanks in advance.

**Comment by wyli** [comment, marked as answer]
Perhaps this is addressed last week via https://github.com/Project-MONAI/MONAI/pull/3841.
